### PR TITLE
feat(template): Change basic project schema to use pipeline

### PR DESCRIPTION
**Pull request by wpietrucha**
# Info
This updates CDK App project template to use pipelines. This is how it looks like in IDE:
image

# Tests
To test it, I've generated project and ensure that newly created was correctly generated and build (`projen build`).

### CODE REVIEW

File: test/__snapshots__/clickup-cdk.test.ts.snap [hunk 1]
```diff
@@ -265,37 +265,130 @@ Object {
...
+}
+
+interface AddStageProps {
+  /**
+   * Where should we add the stage?
+   *
+   * @default githubPipeline
+   */
+  to?: IHasAddStage;
+}
```
Comment (ahammond) [inline test/__snapshots__/clickup-cdk.test.ts.snap:318]: This pattern predated `v4` of `cdk-library`, it's not wrong, but we should probably use the newer approach. Also a good candidate for moving to `cdk-library/cdk-pipeline`.

File: test/__snapshots__/clickup-cdk.test.ts.snap [hunk 2]
```diff
@@ -265,37 +265,130 @@ Object {
...
+        const stage = new Stage(this, stageId, rootId, {
+          ...commonProps,
+          clickUpEnvironment: namedEnv.organizationalUnit, // clickUpEnvironment kinda just duplicated OU. Might as well make that formal.
+          clickUpRole: core.ClickUpRole.APP,
+          namedEnv,
+        });
+        (addStageProps?.to ?? githubPipeline).addStage(stage);
+        return stage;
+      });
+    };
```
Comment (ahammond) [inline test/__snapshots__/clickup-cdk.test.ts.snap:350]: Also something I'd love to see abstracted and moved to `cdk-library/cdk-pipeline`. I think we have enough examples of how we use this stuff by now to do a reasonable job of abstracting it to the library.

**Review by ahammond — COMMENTED** [review]
I love the initiative here. I think that some of this code should be moved into `cdk-library` now that we've seen enough examples to have a pretty good idea of how it should look. Please also remember that with sample code there can be surprising behavior when you delete files (IIRC when you delete all the files in a sample directory they magically re-appear). So, we want to keep our sample code footprint small.

File: src/clickup-cdk.ts [hunk 3]
```diff
@@ -143,36 +143,119 @@ export module clickupCdk {
...
+    });
+    const githubPipeline = new cdkPipeline.GitHubPipeline(this, stackId, {
+      repoString: \`time-loop/\${rootId.addSuffix(['cdk']).kebab}\`,
+    });
+
+    const addStage = (
+      namedEnvFactory: core.INamedEnvFactory,
+      regions: string[],
+      addStageProps?: AddStageProps,
+    ): Stage[] => {
```
Comment (Justinon) [inline src/clickup-cdk.ts:207]: This is great, I love the idea of finally introducing sample pipeline code to the projects by default. This is a tad out of date, we've somewhat recently released `cdk-library` v4 which includes some breaking changes here. What it really does it allow us to deploy to shards dynamically, meaning one will no longer have to explicitly define shards to deploy to, but rather use a filter that the `cdk-library` uses to perform a shard lookup instead.

I'd love for that change to be included here as well. There are quite a few examples out in the wild already, and it wouldn't be much of a stretch to get it implemented here. In fact, I've got a [merged PR here](https://github.com/time-loop/view-service-cdk/pull/27) that takes you from nearly exactly where you are to where we want to be. I highly recommend you look at the commits to sort together the steps to take.

Comment (Justinon) [reply src/clickup-cdk.ts:207]: Ooooooof, this raises a problem actually. This is an open source repo, but as a result of this change we're introducing the inclusion of private package `cdk-library`...effectively defeating the purpose of this repo being open source.

Comment (Justinon) [reply src/clickup-cdk.ts:207]: It's for exactly this reason that I've hesitated on an approach to an abstracted pipeline. If anything, we may want to create a separate (and private) repo that extends these classes to utilize our private stuff.

**Pull request closed without merging**